File: parse_test.js
const fs = require('fs');

let safetyData = {};
try {
  const fileContent = fs.readFileSync('safety-report.json', 'utf8');
  safetyData = JSON.parse(fileContent || '{}');
  console.log('Safety vulns:', safetyData.vulnerabilities.length);
} catch(e) {
  console.error('Failed to parse safety-report.json:', e);
}

File: parse_test.py
import json

def parse_report(filepath):
    try:
        with open(filepath) as f:
            content = f.read()
            if not content.strip():
                return {}
            start = content.find('{')
            end = content.rfind('}')
            if start != -1 and end != -1:
                return json.loads(content[start:end+1])
            return json.loads(content)
    except Exception as e:
        print(f"Error parsing {filepath}: {e}")
        return {}

print("Safety:", parse_report("safety-report.json"))
print("Bandit:", parse_report("bandit-report.json"))
print("Pip-audit:", parse_report("pip-audit-report.json"))
